src/network/client_system.cpp

```diff
@@ -171,20 +155,118 @@ void ClientSystem::Receive( ENetEvent& event )
     //         event.channelID);
     std::istringstream iss( std::string( ( char* )( event.packet->data ), event.packet->dataLength ) );
     eos::portable_iarchive ia( iss );
-    if ( mMessageHolder.GetIncomingMessages().mMessages.empty() )
+
+    MessageList::Messages_t messages;
+    ia >> messages;
+    mMessageHolder.GetIncomingMessages().TransferFrom( messages );
+
+    /* Clean up the packet now that we're done using it. */
+    enet_packet_destroy ( event.packet );
+}
+
+void ClientSystem::UpdateThread()
+{
+    while (mRunning)
```

Comment:
SendMessage calls TransferOutgoingMessagesTo which contains this line:
 mMessageHolder.GetOutgoingMessages().GetCV().wait_for( ulck, std::chrono::milliseconds( mWaitMillisecs ) );
this will sleep for mWaitMillisecs.